### ISSUE TITLE
Wrong diagnostic cfg-name for `unimplemented-builtin-macro`

### ISSUE THREAD

**Issue by boozook** [rust-lang/rust-analyzer#22607]
Currently [`unimplemented-builtin-macro`][] (e.g. for `eii`) reported as [`macro-error`][], but it seems that it should be reported as `unimplemented-builtin-macro`, as well as could be disabled with `initialization options.diagnostics.disabled.unimplemented-builtin-macro`.

Error: `this built-in macro is not implemented`

<details><summary>Full error json</summary>

```json
{"jsonrpc":"2.0","id":100,"result":{"kind":"full","resultId":"rust-analyzer","items":[{"range":{"start":{"line":23,"character":3},"end":{"line":23,"character":6}},"severity":1,"code":"macro-error","codeDescription":{"href":"https://rust-analyzer.github.io/book/diagnostics.html#macro-error"},"source":"rust-analyzer","message":"this built-in macro is not implemented"},{"range":{"start":{"line":27,"character":3},"end":{"line":27,"character":6}},"severity":1,"code":"macro-error","codeDescription":{"href":"https://rust-analyzer.github.io/book/diagnostics.html#macro-error"},"source":"rust-analyzer","message":"this built-in macro is not implemented"}]}}
```

</details>

[`unimplemented-builtin-macro`]: https://rust-analyzer.github.io/book/diagnostics.html#unimplemented-builtin-macro
[`macro-error`]: https://rust-analyzer.github.io/book/diagnostics.html#macro-error
[src]: https://github.com/rust-lang/rust-analyzer/blob/abb1301c3c14a40645bb2588b1cc858fe374b527/crates/hir-expand/src/db.rs#L340

**rust-analyzer version**: `rust-analyzer 1.98.0-nightly (9e2abe0c 2026-06-16)`

**rustc version**: `rustc 1.98.0-nightly (9e2abe0c6 2026-06-16)`

**code snippet to reproduce**:
```rust
#![feature(extern_item_impls)]
#[eii(FOO)]
static FOO: u8;
```

**Comment by ChayimFriedman2** [comment]
It's hard, and I argue you should never disable this diagnostic. If you still want, you can disable `macro-error`.

**Comment by boozook** [comment]
> I argue you should never disable this diagnostic.

But why? Plz give me some explanation. _(excluding the reason like because it doesn't work anyway)_

**Comment by ChayimFriedman2** [comment]
Because the easiest way to emit this error is from the macro itself, and macro-error is the only kind of error a macro can emit.

**Comment by boozook** [comment]
@ChayimFriedman2, thanks for the answer, now I understood.

But how I can disable errors for some unimplemented-builtin-macro like `eii`? I've tried [`procMacro.ignored`][]` with 
```
"core": ["eii", "macros::builtin::eii", "core::macros::builtin::eii"]
```
and it doesn't work for me.


[`procMacro.ignored`]: https://rust-analyzer.github.io/book/configuration.html#procMacro.ignored


- - -

Upd. Okay, now I understand that it's for proc-macro crates only and reexported proc-macros, but built-in macro defined as decl-macro v2 is not "normal proc-macro" definitely. So, how to disable that error without disabling `macro-error` entirely? 😢

**Comment by ChayimFriedman2** [comment]
You can't.
